File: benchmarks/workloads/multi_kind_dispatch/javascript/bench.js
function* ticker(n) {
    for (let i = 0n; i < n; i++) {
        if (i % 2n === 0n) yield ['tick', i];
        else yield ['tock', i];
    }
}

const n = BigInt(process.argv[2]);
let ticks = 0n;
let tocks = 0n;
for (const [kind, _v] of ticker(n)) {
    if (kind === 'tick') ticks++;
    else tocks++;
}
console.log(`ticks = ${ticks} tocks = ${tocks}`);

File: benchmarks/workloads/multi_kind_dispatch/python/bench.py
import sys

def ticker(n):
    i = 0
    while i < n:
        if i % 2 == 0:
            yield ('tick', i)
        else:
            yield ('tock', i)
        i += 1

def main():
    n = int(sys.argv[1])
    ticks = 0
    tocks = 0
    for kind, _v in ticker(n):
        if kind == 'tick':
            ticks += 1
        else:
            tocks += 1
    print(f"ticks = {ticks} tocks = {tocks}")

if __name__ == "__main__":
    main()
